packages/musap-react-native/android/src/main/kotlin/com/sphereon/musap/MusapModule.kt

```diff
@@ -0,0 +1,256 @@
+            }
+            val removedKey = MusapClient.removeKey(musapKey)
+            promise.resolve(removedKey)
+        } catch (e: Exception) {
+            Log.e("MUSAP", "removeKey failed", e)
+            promise.reject(e)
+        }
+    }
+
+
+    @ReactMethod
+    fun sign(req: ReadableMap, promise: Promise) {
+        try {
+            val signatureReq = req.toSignatureReq(this.currentActivity)
+
+            val key = signatureReq.key
+            CoroutineScope(Dispatchers.Default).launch {
+                try {
+                    val result = suspendCancellableCoroutine<MusapSignature> { continuation ->
+                        val musapCallback = object : MusapCallback<MusapSignature> {
+                            override fun onSuccess(signature: MusapSignature?) {
+                                if (signature != null) {
+                                    continuation.resume(signature) {
+                                        Log.w("MUSAP", "Signing cancelled")
+                                    }
+                                } else {
+                                    continuation.resumeWithException(Exception("MusapSignature is null"))
+                                }
+                            }
+
+                            override fun onException(e: MusapException?) {
+                                continuation.resumeWithException(e ?: Exception("sign error"))
+                            }
+                        }
+                        MusapClient.sign(signatureReq, musapCallback)
+
+                        continuation.invokeOnCancellation {
+                            Log.w("MUSAP", "Signing cancelled")
+                        }
+                    }
+                    val convertToRS = true // FIXME to req
+
+                    if (convertToRS) {
+                        val rsSignature = convertDERtoRS(result.rawSignature)
+                        val rsSignatureBase64 = Base64.getUrlEncoder().withoutPadding().encodeToString(rsSignature)
+                        promise.resolve(rsSignatureBase64)
+                    } else {
+                        promise.resolve(result.b64Signature)
+                    }
+                } catch (e: Exception) {
+                    Log.e("MUSAP", "Error signing the data", e)
+                    promise.reject("SIGN_ERROR", "Error signing the data: ${e.message}", e)
+                }
+            }
+        } catch (e: Exception) {
+            Log.e("MUSAP", "sign failed", e)
+            promise.reject("SIGN_ERROR", "Error preparing signature request: ${e.message}", e)
+        }
+    }
+
+    private fun attachSignature(orig: JWSObject, sig: MusapSignature): JWSObject {
+        try {
+            val header = orig.header.toBase64URL()
+            val payload = orig.payload.toBase64URL()
+            val signature = Base64URL.encode(transcodeSignature(sig.rawSignature))
+            return JWSObject(header, payload, signature)
```

Comment:
Where is this used? As we need Cbor/Cose as well, I hope we are not really depending on this

Reply:
It's not, it in a orphan function attachSignature. I ripped JWS support out of the native code

Reply:
I cleaned up the orphaned methods